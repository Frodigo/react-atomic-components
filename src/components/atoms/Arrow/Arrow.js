import React from 'react';
import PropTypes from 'prop-types';

import ArrowIcon from '../../../assets/icons/arrow.svg';

import classes from './Arrow.module.css';

/**
 * This is the arrow.
 * @param {*} props
 */
const Arrow = (props) => {
    const { variant, onClick, ...rest } = props;

    let arrowClass = null;

    switch (variant) {
        case 'right': {
            arrowClass = classes.arrowRight;

            break;
        }

        case 'left': {
            arrowClass = classes.arrowLeft;

            break;
        }

        default: {
            arrowClass = classes.arrowLeft;
        }
    }

    return (
        <button
            type="button"
            className={arrowClass}
            onClick={onClick}
            {...rest}
        >
            <ArrowIcon />
        </button>
    );
};

Arrow.propTypes = {
    disabled: PropTypes.bool,
    onClick: PropTypes.func.isRequired,
    variant: PropTypes.oneOf(['right', 'left'])
};

export default Arrow;
