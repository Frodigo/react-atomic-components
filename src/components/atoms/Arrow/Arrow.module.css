.arrow {
    background: none;
    border: none;
    outline: none;
}

.arrowLeft {
    composes: arrow;
}

.arrowRight {
    composes: arrow;
    transform: rotate(180deg);
}

.arrow[disabled] path {
    fill: var(--color-gray-dark-3);;
}
