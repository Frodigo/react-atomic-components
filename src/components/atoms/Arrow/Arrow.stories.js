import React from 'react';

import Arrow from './Arrow';

export default {
    title: 'Atoms/Controls/Arrow',
    component: Arrow,
    argTypes: {
        onClick: { action: 'clicked' },
        variant: {
            control: {
                type: 'select',
                options: ['right', 'left']
            }
        }
    }
};

const Template = (args) => <Arrow {...args} />;

export const LeftArrow = Template.bind({});
LeftArrow.args = {
    variant: 'left',
    disabled: false
};

export const RightArrow = Template.bind({});
RightArrow.args = {
    variant: 'right',
    disabled: false
};

export const DisabledArrow = Template.bind({});
DisabledArrow.args = {
    variant: 'left',
    disabled: true
};
