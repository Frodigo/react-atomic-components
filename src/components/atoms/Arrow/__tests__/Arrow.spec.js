import React from 'react';
import { render } from '@testing-library/react';

import Arrow from '../Arrow';

describe('Arrow should be rendered', () => {
    const text = 'Call to action';
    const onCLickHandler = jest.fn();

    test('Arrow renders correctly', () => {
        const { container } = render(
            <Arrow variant={'left'} onClick={onCLickHandler}>
                {text}
            </Arrow>
        );

        expect(container).toMatchSnapshot();
    });

    test('Disabled arrow renders correctly', () => {
        const { container } = render(
            <Arrow variant={'right'} onClick={onCLickHandler} disabled>
                {text}
            </Arrow>
        );

        expect(container).toMatchSnapshot();
        expect(container.firstChild.disabled).toBe(true);
    });
});
