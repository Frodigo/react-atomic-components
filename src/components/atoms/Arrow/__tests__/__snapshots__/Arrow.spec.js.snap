// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow should be rendered Arrow renders correctly 1`] = `
<div>
  <button
    class="arrowLeft"
    type="button"
  >
    <span>
      svg
    </span>
  </button>
</div>
`;

exports[`Arrow should be rendered Disabled arrow renders correctly 1`] = `
<div>
  <button
    class="arrowRight"
    disabled=""
    type="button"
  >
    <span>
      svg
    </span>
  </button>
</div>
`;
